sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractAsynchronousInstrument.java

```diff
@@ -11,45 +11,39 @@
 import io.opentelemetry.sdk.metrics.data.MetricData;
 import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
 import java.util.concurrent.locks.ReentrantLock;
 import javax.annotation.Nullable;
 
 abstract class AbstractAsynchronousInstrument<T extends AsynchronousInstrument.Result>
     extends AbstractInstrument implements AsynchronousInstrument<T> {
-  @Nullable private volatile Callback<T> metricUpdater = null;
+  @Nullable private final Callback<T> metricUpdater;
```

Comment:
yes, this was always in my head as a thing that needed to be done. Thanks!

Reply:
That's why I separated from the previous PR, because I wanted to fix this deeper